app/controllers/sessions_controller.rb

```diff
@@ -66,4 +68,34 @@ def destroy
     end
     redirect_to(:root, notice: t('.success'))
   end
+
+  private
+
+  def redirect_to_survey
+    # This method is taken from Xikolo and slightly adapted.
+    # Forward arbitrary optional query params to LimeSurvey
+    # and remove tracking and other sensitive user params.
+    query_params = request
+      .query_parameters
+      .delete_if {|key, _| key.to_s.match(/.*(tracking|referrer|user_id).*/) }
+      .merge(
+        r: 'survey/index', # required LimeSurvey path passed as query param
+        sid: params[:custom_survey_id], # required LimeSurvey survey ID
+        newtest: 'Y', # force a new LimeSurvey session
+        xi_platform: 'openhpi' # pass a platform identifier
+      ).tap do |qp|
+      if current_user
+        # add a user pseudo ID if applicable
+        qp[:xi_pseudo_id] = Digest::SHA256.hexdigest(current_user.external_id)
```

Comment:
It would be nice to also have the study_group_id

Reply:
I've just amended my previous commit and added the `study_group_id` as `co_study_group_id`.